packages/expo-updates/android/src/main/java/expo/modules/updates/loader/ExpoMultipartStreamReader.kt

```diff
@@ -0,0 +1,162 @@
+package expo.modules.updates.loader
+
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString
+import java.io.IOException
+import java.util.HashMap
+import kotlin.math.max
+
+/**
+ * Fork of {@link com.facebook.react.devsupport.MultipartStreamReader} that doesn't necessarily
+ * expect a preamble (first boundary is not necessarily preceded by CRLF).
+ */
+class ExpoMultipartStreamReader(private val source: BufferedSource, private val boundary: String) {
+  private var mLastProgressEvent: Long = 0
```

Comment:
I'm hesitant to edit this stream reader too much. For now I didn't read it too closely but did read through its test case, but now that you mention it the fact that we had to fork it is probably a pretty good case for needing to read through it more, even though only one line needed modification.

For reference, the other multipart readers I tried were:
- https://github.com/synchronoss/nio-multipart - needed some odd async parsing stuff but now I'm thinking it may be better
- might be able to use https://james.apache.org/mime4j/index.html, but it is fairly specifically designed for email.
- https://commons.apache.org/proper/commons-fileupload/apidocs/org/apache/commons/fileupload/MultipartStream.html - going to look more into this.